functions/_common/explorerHandlers.ts

```diff
@@ -0,0 +1,216 @@
+import ReactDOMServer from "react-dom/server"
+import { Env, extensions } from "./env.js"
+import { extractOptions, ImageOptions } from "./imageOptions.js"
+import {
+    buildExplorerProps,
+    Explorer,
+    ExplorerChoiceParams,
+} from "@ourworldindata/explorer"
+import { renderSvgToPng } from "./grapherRenderer.js"
+import { error, png } from "itty-router"
+import { createZip, File } from "littlezipper"
+import { Bounds, Url } from "@ourworldindata/utils"
+import {
+    getSelectedEntityNamesParam,
+    GrapherState,
+    migrateSelectedEntityNamesParam,
+    SelectionArray,
+} from "@ourworldindata/grapher"
+import {
+    assembleCsv,
+    assembleReadme,
+    ensureDownloadOfDataAllowed,
+} from "./downloadFunctions.js"
+import { assembleMetadata } from "./metadataTools.js"
+
+async function initGrapherForExplorerView(
+    env: Env,
+    options: ImageOptions
+): Promise<{
+    grapherState: GrapherState
+    explorerParams: ExplorerChoiceParams
+}> {
+    const url = env.url
+    const explorerPage = await env.ASSETS.fetch(url, { redirect: "manual" })
+
+    const html = await explorerPage.text()
+    const queryStr = url.searchParams.toString()
+    // The env URL class isn't compatible with the Url class from @ourworldindata/utils
+    const urlObj = Url.fromURL(url.toString())
+    const [windowEntityNames] = [urlObj]
+        .map(migrateSelectedEntityNamesParam)
+        .map(getSelectedEntityNamesParam)
+
+    const selection = new SelectionArray(windowEntityNames)
+    const bounds = new Bounds(0, 0, options.svgWidth, options.svgHeight)
+    const explorerProps = await buildExplorerProps(
+        html,
+        queryStr,
+        selection,
+        bounds
+    )
+    const explorer = new Explorer(explorerProps)
+    explorer.updateGrapherFromExplorer()
```

Comment:
Similarly, here: I know this is existing code, but could we make this awaitable and get rid of the timeout? If not, it would be good to have a code comment explaining why we have to do this.

Reply:
Nope, sorry. I imagine I tried to make it return a promise and encountered some obstacle where it broke other parts of the site, but I don't remember the specifics. 😑